widget/fileicon.go

```diff
@@ -161,9 +166,11 @@ func (s *fileIconRenderer) Refresh() {
 	if s.file.URI != s.file.cachedURI {
 		s.file.propertyLock.RLock()
 		s.file.setURI(s.file.URI)
-		s.updateObjects()
 		s.file.cachedURI = s.file.URI
 		s.file.propertyLock.RUnlock()
+
+		s.img.Resource = s.file.resource
+		s.ext.Text = s.file.extension
```

Comment:
Should these not be read with the read lock?

Reply:
Good catch thanks